Refresh token

Korkkii opened the pull request:
Adds automatic refreshing of access token when it expires.

Commit:
Add automatic refresh for access token if expired

Commit:
Add README instuctions for access token flow

pihvi commented on examples/access_token.js:
```diff
@@ -0,0 +1,39 @@
+// Initialize Google auth middleware. You need your Google app id and secret.
+const myGauth = gauth({
+  clientID: '384494927189-7ls0vh3mnsoc94s0q2dl5p21rdf9g82f.apps.googleusercontent.com',
+  clientSecret: 'C9YCedLkvkm_mml_L5Jizd-p',
```
Bit worrying to see secrets added in public commit..

rjoonas commented on package.json:
```diff
@@ -13,8 +13,10 @@
   },
   "homepage": "https://github.com/reaktor/express-gauth#readme",
   "dependencies": {
+    "moment": "^2.20.1",
```
I personally prefer [date-fns](https://date-fns.org/) over moment.

Commit:
Add more explicit failing for token refresh flow

rjoonas commented on README.md:
```diff
@@ -90,12 +110,17 @@ gauth({
+    accessType: 'offline',
+    // This forces Google to send refresh tokens on each session start
+    // By default Google sends refresh token only once per authentication to Google
+    prompt: 'consent',
```
trailing comma

Commit:
Add option to specify refresh window

Commit:
Add forced relogin if missing refresh token and trying to refresh access token

Korkkii commented:
Added a forced relogin if trying to refresh access token when missing the refresh token. 

This can happen in a case where user starts a new session with a service, but user has previously authorized the service thus no consent authorization happens. Google sends the refresh token only on the first time authorizing a service, so in this case it won't send it. This forces a new consent screen which results in refresh token sent and added to user's session.

rjoonas commented on index.js:
```diff
@@ -44,7 +54,33 @@ module.exports = function expressGAuth(options) {
+              const newGoogleParams = Object.assign(googleAuthorizationParams,
+                { prompt: 'consent' })
+              const newConfig = Object.assign(config,
+                { googleAuthorizationParams: newGoogleParams })
```
Why bind `newConfig` at all if the purpose is to just mutate `config`? Can't see `newConfig` being used anywhere.

rjoonas commented:
I actually need this feature in another project so it would be good to get this in. Currently using this branch for local development and it seems to work.

pihvi commented:
So make a release of this?

Commit:
Refactor out mutation in shouldBeRefreshed branch
